source/intel/asm/crc32c_sse42_asm.c

```diff
@@ -6,6 +6,12 @@
 #include <aws/checksums/private/crc_priv.h>
 
 #include <aws/common/cpuid.h>
+#include <emmintrin.h>
```

Comment:
`mv source/intel/asm source/intel/intrin`

then we need to see if the visualc code will actually work because i think that's just using the other intrinsics. Also, I think we can just delete the visualc code if we're not using any asm. Can we delete visualc and have cmake use this file for both and see what happens?

Reply:
Results moving the asm folder to intrin
mv source/intel/asm source/intel/intrin
[root@aws-spr4 build]# make
[ 16%] Building C object CMakeFiles/aws-checksums.dir/source/crc.c.o
[ 33%] Building C object CMakeFiles/aws-checksums.dir/source/crc_sw.c.o
[ 50%] Linking C static library libaws-checksums.a
[ 50%] Built target aws-checksums
[ 66%] Building C object tests/CMakeFiles/aws-checksums-tests.dir/test_runner.c.o
[ 83%] Building C object tests/CMakeFiles/aws-checksums-tests.dir/crc_test.c.o
[100%] Linking C executable aws-checksums-tests
/usr/bin/ld: ../libaws-checksums.a(crc.c.o): in function `aws_checksums_crc32':
crc.c:(.text+0x41): undefined reference to `aws_checksums_crc32_hw'
/usr/bin/ld: ../libaws-checksums.a(crc.c.o): in function `aws_checksums_crc32c':
crc.c:(.text+0xa1): undefined reference to `aws_checksums_crc32c_hw'
collect2: error: ld returned 1 exit status
make[2]: *** [tests/CMakeFiles/aws-checksums-tests.dir/build.make:115: tests/aws-checksums-tests] Error 1
make[1]: *** [CMakeFiles/Makefile2:882: tests/CMakeFiles/aws-checksums-tests.dir/all] Error 2
make: *** [Makefile:146: all] Error 2



Removing source/intel/visualc folder 

[root@aws-spr4 build]# make
[ 14%] Building C object CMakeFiles/aws-checksums.dir/source/crc.c.o
[ 28%] Building C object CMakeFiles/aws-checksums.dir/source/crc_sw.c.o
[ 42%] Building C object CMakeFiles/aws-checksums.dir/source/intel/asm/crc32c_sse42_asm.c.o
[ 57%] Linking C static library libaws-checksums.a
[ 57%] Built target aws-checksums
[ 71%] Building C object tests/CMakeFiles/aws-checksums-tests.dir/test_runner.c.o
[ 85%] Building C object tests/CMakeFiles/aws-checksums-tests.dir/crc_test.c.o
[100%] Linking C executable aws-checksums-tests
[100%] Built target aws-checksums-tests
[root@aws-spr4 build]# make install
Consolidate compiler generated dependencies of target aws-checksums
[ 57%] Built target aws-checksums
Consolidate compiler generated dependencies of target aws-checksums-tests
[100%] Built target aws-checksums-tests
Install the project...
-- Install configuration: "Release"
-- Installing: /usr/local/lib64/libaws-checksums.a
-- Up-to-date: /usr/local/include/aws/checksums/crc.h
-- Up-to-date: /usr/local/include/aws/checksums/exports.h
-- Installing: /usr/local/lib64/aws-checksums/cmake/static/aws-checksums-targets.cmake
-- Installing: /usr/local/lib64/aws-checksums/cmake/static/aws-checksums-targets-release.cmake
-- Installing: /usr/local/lib64/aws-checksums/cmake/aws-checksums-config.cmake

Reply:
my suspicion on the build failure by moving asm to intrin and updating cmake to find it and it still not building is the cmake cache. I usually just wipe out the build and re-run cmake to make sure it's not that.

The visualc stuff would have to be run on an msvc compiler to check. I have my windows machine up and i'll give it a look and report back.

Reply:
Regarding move asm to intrin. Clean it again and rebuild it with no issues.